cicecore/cicedyn/infrastructure/io/io_pio2/ice_history_write.F90

```diff
@@ -1341,25 +1240,107 @@ subroutine ice_write_hist (ns)
 
       end subroutine ice_write_hist
 
+
 !=======================================================================
 
-      subroutine ice_write_hist_attrs(File, varid, hfield, ns)
+   ! Defines a coordinate var in the history file
+      subroutine ice_hist_coord_def(File, coord,lprecision, dimids,varid)
 
+      use pio, only: file_desc_t, var_desc_t, pio_def_var,  pio_put_att
+#ifndef USE_PIO1
+      use pio, only: pio_def_var_deflate
+      use pio_nf, only: pio_def_var_chunking !This is missing from pio module <2.6.0
+      use netcdf, only: NF90_CHUNKED
+      use ice_history_shared, only: history_deflate, history_chunksize, history_format
+#endif
+      
+      type(file_desc_t), intent(in) :: File
+      type(coord_attributes), intent(in) :: coord
+      integer(kind=int_kind), intent(in) :: dimids(:), lprecision
+      type(var_desc_t), intent(inout):: varid
+      
+      ! local vars
+      integer(kind=int_kind) :: chunks(size(dimids)), i, status
+
+      character(len=*), parameter :: subname = '(ice_hist_coord_def)'
+
+      !define var, set deflate, long_name and units
+      status = pio_def_var(File, coord%short_name, lprecision, dimids, varid)
+      call ice_pio_check(status, &
+         subname//' ERROR: defining coord '//coord%short_name,file=__FILE__,line=__LINE__)
+#ifndef USE_PIO1
+      if (deflate==1 .and. size(dimids)>1 ) then 
+         status = pio_def_var_deflate(File, varid, shuffle, deflate ,history_deflate)
+         call ice_pio_check(status, &
+            subname//' ERROR: deflating coord '//coord%short_name,file=__FILE__,line=__LINE__)
+      endif
+
+      if (history_format=='hdf5' .and. dimids(1)==imtid .and. dimids(2)==jmtid) then 
```

Comment:
I need to revisit this for 1-d coords

Reply:
Yeah - the advice seems to be don't chunk the 1d vars. Most coords use i as the first var in the array and j as the second, but there are a couple of exceptions (which we can basically ignore / not chunk).